lambda_functions/v1/openapi/lpa-codes-openapi-v1.yml

```
@@ -145,7 +145,7 @@ paths:
                     properties:
                       lpa:
                         type: string
-                        pattern: '[0-9]{12}|^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$'
+                        pattern: '([0-9]{12}|^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12})|(M(-[0-9A-Z]{4}){3})$'
```

Needs the `^` moving and an extra set of brackets or it will accept extra characters at the beginning or end of the ID

```suggestion
                        pattern: '^([0-9]{12}|([a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12})|(M(-[0-9A-Z]{4}){3}))$'
```